File: array/1854_maximum_population_year/1854_maximum_population_year.py
from typing import List

class Solution:
    def maximumPopulation(self, logs: List[List[int]]) -> int:
        nums = [0] * 2051
        ret = 1950
        for i in logs:
            for j in range(i[0], i[1]):
                nums[j] += 1
        for x in range(1950, len(nums)):
            if nums[x] > nums[ret]:
                ret = x
        return ret


logs = [[1993,1999],[2000,2010]]
res = Solution().maximumPopulation(logs)
print(res)

File: array/1854_maximum_population_year/1854_maximum_population_year.go
package main

import "fmt"

func maximumPopulation(logs [][]int) int {
	var year [101]int
	for _, l := range logs {
		year[l[0]-1950]++
		year[l[1]-1950]--
	}
	cnt := 0
	mcnt := 0
	ret := 0
	for y, c := range year {
		cnt += c
		if cnt > mcnt {
			mcnt = cnt
			ret = y + 1950
		}
	}
	return ret
}

func main() {
	logs := [][]int{{1993, 1999}, {2000, 2010}}
	res := maximumPopulation(logs)
	fmt.Println(res)
}
